athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/RecordBatchSerDe.java

```diff
@@ -33,7 +34,10 @@
 
 /**
  * used to serialize and deserialize ArrowRecordBatch.
+ *
+ * @deprecated {@link ArrowRecordBatchSerDe} should be used instead
  */
+@Deprecated
 public class RecordBatchSerDe
```

Comment:
In what way is it not backwards compatible? I'm generally unclear on what it means if someone keeps using this old class. My gut, keeping in mind I don't fully understand what you mean here yet, is that we should not have this class in the GA version of the SDK.

Reply:
There are some incompatibilities between the old ObjectMapper and the new one: some classes used custom type strings, eg. "sortable" for SortedRangeSet, removal of redundant fields, and slight inconsistencies in schema/block serialization.  I've verified that Ping is backwards compatible so when Athena receives the response from old (current) SDKs, it can/will use the old ObjectMapper to continue communication.

Another option is to move the old ObjectMapper to Athena and remove it from the SDK.  What do you think of this option?

Reply:
I'm not sure I have a view on moving the old ObjectMapper, if no one can choose to use it directly (e.g. only gets used as the result of a PingResponse) then either is fine with me.